### DISCUSSION TITLE
How to test file upload

### DISCUSSION THREAD

**Discussion by rajeshgautamphd** [antiwork/shortest#201] [Q&A]
The forms testing requires to upload a file. I get error:
    Cannot directly access local file system for security reasons - file upload requires manual interaction with system file picker.
When I gave local file url in the prompt file:///Users/... it gives the same error.

How can I get the file submissions to work?

Thanks

**Comment by slavingia** [comment]
I'm not sure if this is currently possible, but @m2rads may have looked into this.

**Comment by rajeshgautamphd** [comment]
Thanks for following up. I am not a browser technology expert nor a TS expert. Is there a way to modify the the browser dom with the local file path without having to pop up/click the file picker. Alternatively, if we can read and load data from a public url instead from a file.
An overlapping project skyvern does it. Will investigate how  they do it.

    On Saturday, December 28, 2024 at 05:44:58 PM EST, Sahil Lavingia ***@***.***> wrote:  
 
 


I'm not sure if this is currently possible, but @m2rads may have looked into this.

**Reply by nazargladish** [reply]
@rajeshgautamphd Hi [Rajesh](https://github.com/rajeshgautamphd), could you provide an example of form you're trying to upload file to and then this error is thrown? I will take a look

**Reply by rajeshgautamphd** [reply]
created a pull request by adding changes to upload file. tested

**Reply by rajeshgautamphd** [reply]
https://github.com/anti-work/shortest/pull/271
